include/rocksdb/comparator.h

```diff
@@ -78,6 +78,12 @@ class Comparator {
   // The major use case is to determine if DataBlockHashIndex is compatible
   // with the customized comparator.
   virtual bool CanKeysWithDifferentByteContentsBeEqual() const { return true; }
+
+  virtual size_t TimestampSize() const { return 0; }
+
+  virtual int CompareWithoutTimestamp(const Slice& a, const Slice& b) const {
+    return Compare(a, b);
```

Comment:
Since it is a double virtual function call. Maybe we should directly implement it in BytewiseComparator.

Reply:
If performance is a concern, then yes. We implement both `Compare` and `CompareWithoutTimestamp` directly.

Reply:
Please do that in BytewiseComparator and ReverseBytewiseComparator. I an worried about performance regression.

Reply:
Done.